src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -572,6 +573,66 @@ export const selectEarnUserStakingOpportunityByUserStakingId = createDeepEqualOu
   },
 )
 
+// A staking opportunity by opportunityId parsed as an EarnOpportunityType
+// TODO: testme
+export const selectEarnUserStakingOpportunityByOpportunityId = createDeepEqualOutputSelector(
+  selectUserStakingOpportunitiesById,
+  selectStakingOpportunitiesById,
+  selectOpportunityIdParamFromFilter,
+  selectUserStakingIdParamFromFilter,
+  selectMarketDataUserCurrency,
+  selectAssets,
+  (
+    userStakingOpportunitiesById,
+    stakingOpportunitiesById,
+    opportunityId,
+    userStakingId,
+    marketData,
+    assets,
+  ): StakingEarnOpportunityType | undefined => {
+    if (!userStakingOpportunitiesById || !stakingOpportunitiesById || !marketData || !opportunityId)
+      return
+
+    const userStakingOpportunity = userStakingId
+      ? userStakingOpportunitiesById[userStakingId]
+      : undefined
+    const stakingOpportunity = stakingOpportunitiesById[opportunityId]
+
+    if (!stakingOpportunity) return
+
+    const asset = assets[stakingOpportunity.assetId]
+    const underlyingAsset = assets[stakingOpportunity.underlyingAssetId]
+
+    const marketDataPrice = marketData[asset?.assetId ?? underlyingAsset?.assetId ?? '']?.price
+
+    const earnUserStakingOpportunity: StakingEarnOpportunityType = {
+      ...userStakingOpportunity,
+      ...stakingOpportunity,
+      isLoaded: userStakingOpportunity?.isLoaded ?? true,
+      chainId: fromAssetId(stakingOpportunity.assetId).chainId,
+      cryptoAmountBaseUnit: userStakingOpportunity?.stakedAmountCryptoBaseUnit ?? '0',
+      cryptoAmountPrecision: bnOrZero(userStakingOpportunity?.stakedAmountCryptoBaseUnit)
+        .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision ?? 1))
+        .toFixed(),
+      fiatAmount: bnOrZero(userStakingOpportunity?.stakedAmountCryptoBaseUnit)
+        .div(bn(10).pow(bnOrZero(asset?.precision ?? underlyingAsset?.precision)))
+        .times(marketDataPrice ?? '0')
+        .toString(),
+      stakedAmountCryptoBaseUnit: userStakingOpportunity?.stakedAmountCryptoBaseUnit ?? '0',
+      opportunityName: stakingOpportunity.name,
+      icons: makeOpportunityIcons({ opportunity: stakingOpportunity, assets }),
+      contractAddress: isFoxEthStakingAssetId(stakingOpportunity.assetId)
+        ? fromAssetId(stakingOpportunity.assetId).assetReference
+        : undefined,
+      rewardAddress: isFoxEthStakingAssetId(stakingOpportunity.assetId)
+        ? fromAssetId(foxAssetId).assetReference
+        : undefined,
+    }
+
+    return earnUserStakingOpportunity
+  },
+)
+
```

Comment:
ser

Reply:
Fixed in [8320137](https://github.com/shapeshift/web/pull/7952/commits/8320137be514cbc346930b96b3bd16f37d62b146)